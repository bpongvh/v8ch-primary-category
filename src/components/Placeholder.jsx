const { Component } = wp.element;

class Placeholder extends Component {
	static getDerivedStateFromProps( nextProps, prevState ) {
		const canSetPrimaryCategoryName = nextProps.primaryCategoryId &&
			! nextProps.categories.isLoading && nextProps.categories.data;
		if ( canSetPrimaryCategoryName ) {
			const primaryCategory = nextProps.categories.data
				.find( category => {
					return category.id.toString() === nextProps.primaryCategoryId;
				} );
			return { primaryCategoryName: primaryCategory.name };
		}
		return prevState;
	}
	constructor() {
		super( ...arguments );
		this.state = { primaryCategoryName: 'Unknown' };
	}

	render() {
		return (
			<p className="placeholder">In Primary Category: { this.state.primaryCategoryName }</p>
		);
	}
}

export default Placeholder;
