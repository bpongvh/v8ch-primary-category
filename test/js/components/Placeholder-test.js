import { shallow } from 'enzyme';
import Placeholder from '../../../src/components/Placeholder';

const categories = { data: [ { id: 2, name: 'Test Category' } ] };

describe( 'Placeholder', () => {
	test( 'should render', () => {
		const rendered = shallow(
			<Placeholder
				categories={ categories }
				primaryCategoryId="2"
			/>
		);
		expect( rendered ).toMatchSnapshot();
	} );
} );
