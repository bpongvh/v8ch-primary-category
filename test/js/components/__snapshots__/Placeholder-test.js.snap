// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder should render 1`] = `
<p
  className="placeholder"
>
  In Primary Category: 
  Test Category
</p>
`;
